### PR TITLE
docs: Add requirement for bitcoind.

### PR DESCRIPTION
**Pull request by ziggie1984**
Lightning needs witness data, but bitcoind allows for a setting where all rpc commands are stripped of this data. I think we can only mention this in the docs but cannot really check otherwise?

Refers to: #8028

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Thanks, LGTM :tada:

File: docs/INSTALL.md [hunk 1]
```diff
@@ -458,6 +458,11 @@ the following:
   to provide a reliable delivery of notifications (e.g.
   `--zmqpubrawblock=tcp://127.0.0.1:28332` and
   `--zmqpubrawtx=tcp://127.0.0.1:28333`).
+- Make sure you do NOT use the bitcoind config setting `--rpcserialversion`.
```
Comment (yyforyongyu) [inline docs/INSTALL.md:461]:
```suggestion
- Make sure the config `--rpcserialversion` in `bitcoind` is NOT used. 
```

File: docs/INSTALL.md [hunk 2]
```diff
@@ -458,6 +458,11 @@ the following:
...
+  Lightning depends on segwit transactions and the default rpc bitcoind
+  behaviour includes the witness data. Changes to this setting are safety
+  critical and might lead to loss of funds.
+
```
Comment (yyforyongyu) [inline docs/INSTALL.md:465]: Maybe we can also link the discussion here?
https://github.com/bitcoin/bitcoin/issues/28730

File: docs/INSTALL.md [hunk 3]
```diff
@@ -458,6 +458,11 @@ the following:
   to provide a reliable delivery of notifications (e.g.
   `--zmqpubrawblock=tcp://127.0.0.1:28332` and
   `--zmqpubrawtx=tcp://127.0.0.1:28333`).
+- Make sure you do NOT use the bitcoind config setting `--rpcserialversion`.
+  Lightning depends on segwit transactions and the default rpc bitcoind
```
Comment (yyforyongyu) [inline docs/INSTALL.md:462]: We should mention this `-rpcserialversion` must be set to 1 and if we don't explicitly set it in the config file we are fine as it's already default to 1.

Also feel like there's no need to explain in details other than when setting config value to 0 the witness data won't be returned.

**Review by yyforyongyu — COMMENTED** [review]
Thanks for the PR! Yeah don't think we can easily check `bitcoind` config values in `lnd`.

File: docs/INSTALL.md (hunk 1 above)
Comment (guggero) [reply docs/INSTALL.md:461]: Ah, I missed this before, but the config flags in `bitcoind` only have a single dash.

**Review by yyforyongyu — APPROVED** [review]
LGTM🙏

**Pull request merged**